common/recommender/Baseline.js

```diff
@@ -317,6 +318,24 @@ class Baseline {
 
     return entries;
   }
+
+  _dedupeSites(entries) {
+    let dedupedEntries = new Map();
+    entries.forEach(item => {
+      let key = this._createDedupeKey(item);
+      if (!dedupedEntries.get(key)) {
+        dedupedEntries.set(key, [item]);
+      }
+    });
+
+    let results = [];
+    dedupedEntries.forEach(item => results.push(item[0]));
```

Comment:
they come in sorted order, so the first one we encounter will already have the greatest weight